client/client.go

```diff
@@ -233,9 +261,32 @@ func (c *httpClusterClient) reset(eps []string) error {
 		neps[i] = *u
 	}
 
-	c.endpoints = shuffleEndpoints(c.rand, neps)
-	// TODO: pin old endpoint if possible, and rebalance when new endpoint appears
-	c.pinned = 0
+	switch c.selectionMode {
+	case EndpointSelectionRandom:
+		c.endpoints = shuffleEndpoints(c.rand, neps)
+		c.pinned = 0
+	case EndpointSelectionPrioritizeLeader:
+		c.endpoints = neps
+		// TODO: should return ErrNoEndpoints in a case of getting leader fail?
+		lep, err := c.getLeaderEndpoint()
+		if err != nil {
+			return ErrNoEndpoints
```

Comment:
I'd really rather not, can we create a new error here? I also don't like completely masking whatever is going wrong in getLeaderEndpoint